cdap-common/src/main/java/io/cdap/cdap/common/http/AuthenticationChannelHandler.java

```diff
@@ -120,6 +127,22 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
 
     try {
       ctx.fireChannelRead(msg);
+    } finally {
+      ctx.write(msg);
```

Comment:
I am not sure you can write the request back.

Reply:
I think this is no longer needed as we are covering both WRITE and CLOSE